### PR TITLE
added onMsaCode to ClientOptions

### PR DESCRIPTION
**Pull request by Jarco-dev**
The onMsaCode parameter was already implemented but not yet documented, this pr fixes that

### CODE REVIEW

**Comment by Jarco-dev** [comment]
(sorry for the random removal of spaces on some lines my ide does it automatically if this is an issue I'll revert it)

File: index.d.ts [hunk 1]
```diff
@@ -47,6 +47,8 @@ declare module "bedrock-protocol" {
     realms?: RealmsOptions
     // the path to store authentication caches, defaults to .minecraft
     profilesFolder?: string | false
+    // Called when microsoft authorization is needed when not provided it will the information log to the console instead
+    onMsaCode?: (data: MicrosoftDeviceAuthorizationResponse) => void;
```
Comment (LucienHH) [inline index.d.ts]: Could we import the types from prismarine-auth (ServerDeviceCodeResponse) in case they're changed in the future

Comment (Jarco-dev) [reply index.d.ts]: Yes, I didn't realise it was being exported. I Just noticed it being declared the same way in node-minecraft-protocol.

I just pushed a commit changing it.

**Comment by extremeheat** [comment]
Thanks!

**Pull request merged**